src/App.jsx

```diff
@@ -2,35 +2,50 @@ import * as React from "react";
 import TodoList from "./TodoList"; //Import TodoList component
 import AddTodoForm from "./AddTodoForm"; //Import AddTodoForm component
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem("savedTodoList")) || []
-  );
+function App() {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true)
 
   React.useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
+          },
```

Comment:
in this code   data: {
            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
          }, you assign to resolve.data  todoList: JSON.parse(localStorage.getItem("savedTodoList")), you shouldn't use `|| []` here, because useState here ` const [todoList, setTodoList] = React.useState([]);` assign empty array as todoList, and in previous useEffect (lines 27-31) you are creating in localStorage variable `savedTodoList ` and set it to todoList: `localStorage.setItem('savedTodoList', JSON.stringify(todoList))`.